protocols/autonat/src/v2/client/behaviour.rs

```diff
@@ -374,22 +374,6 @@ impl Default for Behaviour<OsRng> {
     }
 }
 
-pub struct Error {
-    pub(crate) inner: dial_request::DialBackError,
-}
-
-impl Display for Error {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        Display::fmt(&self.inner, f)
-    }
-}
-
-impl Debug for Error {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        Debug::fmt(&self.inner, f)
-    }
-}
```

Comment:
Wouldnt it be better to leave this in place but implement `Error` (or maybe use `thiserror::Error` and use `#[error(transparent)]`) so you can downcast to `DialBackError`?

Reply:
I honestly give that a small thought initially. I definitely like the second approach more. But I fail to identify a benefit from that layer of indirection, so came with this approach; do you have some on your mind? Like IIRC the idea of `thiserror` is that someone downstream who uses it can wrap this easily for their needs, and we don't imply it on those who don't want to use it. 😆